tests/integration/Core/Content/Category/SalesChannel/CategoryRouteTest.php

```diff
@@ -2,57 +2,75 @@
 
 namespace Shopware\Tests\Integration\Core\Content\Category\SalesChannel;
 
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
 use PHPUnit\Framework\Attributes\Group;
 use PHPUnit\Framework\TestCase;
 use Shopware\Core\Content\Category\Exception\CategoryNotFoundException;
+use Shopware\Core\Content\Category\SalesChannel\CategoryRoute;
 use Shopware\Core\Content\Product\Aggregate\ProductVisibility\ProductVisibilityDefinition;
 use Shopware\Core\Defaults;
 use Shopware\Core\Framework\Context;
 use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\Log\Package;
 use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
 use Shopware\Core\Framework\Test\TestCaseBase\SalesChannelApiTestBehaviour;
 use Shopware\Core\Test\Stub\Framework\IdsCollection;
+use Shopware\Tests\Integration\Core\Content\Category\SalesChannel\fixtures\CategoryRouteInheritanceFixtures;
 use Symfony\Bundle\FrameworkBundle\KernelBrowser;
 
 /**
  * @internal
+ *
+ * @phpstan-type CmsInheritanceDataProviderActual array{
+ *      activeLanguageCode: string,
+ *      hasTemplate: list<string>,
+ *      hasSlotOverride: list<string>,
+ *  }
+ * @phpstan-type CmsInheritanceDataProviderIterator iterable<array{
+ *       actual: CmsInheritanceDataProviderActual,
+ *       expected: string
+ *  }>
  */
 #[Group('store-api')]
+#[Package('discovery')]
+#[CoversClass(CategoryRoute::class)]
 class CategoryRouteTest extends TestCase
 {
     use IntegrationTestBehaviour;
     use SalesChannelApiTestBehaviour;
 
+    private const LANGUAGE_IDS = [
+        'en' => Defaults::LANGUAGE_SYSTEM,
+        'de' => '20354d7ae4fe47af8ff6187bc0dedede',
+        'at' => '20354d7ae4fe47af8ff6187bc0aaaaaa',
+    ];
+
     private KernelBrowser $browser;
 
     private IdsCollection $ids;
 
     protected function setUp(): void
     {
         $this->ids = new IdsCollection();
-
-        $this->createData();
-
-        $this->browser = $this->createCustomSalesChannelBrowser([
-            'id' => $this->ids->create('sales-channel'),
-            'navigationCategoryId' => $this->ids->get('home-category'),
-        ]);
-
-        $this->setVisibilities();
     }
 
     public function testCmsPageResolved(): void
     {
+        $this->createListingData();
```

Comment:
why not in the setUp method?

Reply:
Because the data is depending on the case. My Tests do not work with ListingData, therefore I pushed those to the cases.